docs/spec/spec-proposals/f1-fee-distribution/f1_fee_distr.tex

```diff
@@ -17,49 +17,69 @@
-Then this entry is created using the previous entry as follows: $$\sum_{i=0}^{f}\frac{T_i}{n_i} = \sum_{i=0}^{f-1}\frac{T_i}{n_i} + \frac{T_f}{n_f} = entry_{f-1} + \frac{T_f}{n_f}$$
+Let $p_0$ represent the period which begins when the validator first bonds.
+The central idea to the F1 model is that at the end of the $k$th period,
+the following is stored at a state location indexable by $k$: $\sum_{i=0}^{k}\frac{T_i}{n_i}$.
+Let the index of the current period be $f$. 
+When a delegator wants to delegate or withdraw their reward, they first create a new entry in state to end the current period.
+Then this entry is created using the previous entry as follows:
+$$Entry_f = \sum_{i=0}^{f}\frac{T_i}{n_i} = \sum_{i=0}^{f-1}\frac{T_i}{n_i} + \frac{T_f}{n_f} = Entry_{f-1} + \frac{T_f}{n_f}$$
 Where $T_f$ is the fees the validator has accrued in period $f$, and $n_f$ is the validators total amount of stake in period $f$.
 
-The withdrawer's delegation object has the index $k$ for the period which they started accruing fees for.
-Thus the reward they should receive when withdrawing is:
+The withdrawer's delegation object has the index $k$ for the period which they ended by bonding. (They start receiving rewards for period $k + 1$)
+The reward they should receive when withdrawing is:
 
-$$x\left(entry_f - entry_k\right)  = x\left(\left(\sum_{i=0}^{f}\frac{T_i}{n_i}\right) - \left(\sum_{i=0}^{k}\frac{T_i}{n_i}\right)\right) = x \sum_{i = k}^{f} \frac{T_i}{n_i}$$
+$$x \sum_{i = k + 1}^{f} \frac{T_i}{n_i} = x\left(\left(\sum_{i=0}^{f}\frac{T_i}{n_i}\right) - \left(\sum_{i=0}^{k}\frac{T_i}{n_i}\right)\right) = x\left(Entry_f - Entry_k\right)$$
 
-The first summation is the state entry for $f$, and the second sum is the state entry at $k$.
-It is clear from the equations that this payout mechanism maintains correctness, and required no iterations.
+It is clear from the equations that this payout mechanism maintains correctness, and required no iterations. It just needed the two state reads for these entries.
 
 $T_f$ is a separate variable in state for the amount of fees this validator has accrued since the last update to its power.
 This variable is incremented at every block by however much fees this validator received that block.
-On the update to the validators power, this variable is used to create the entry in state at $f$.
+On the update to the validators power, this variable is used to create the entry in state at $f$, and is then reset to 0.
 
 This fee distribution proposal is agnostic to how all of the blocks fees are divied up between validators.
-This creates many nice properties, for example only rewarding validators who signed that block.
+This creates many nice properties, for example it is possible to only rewarding validators who signed that block.
  
 \section{Additional add-ons}
 \subsection{Commission Rates}
```

Comment:
This can be generalized, I think, to a function `f :: f_h -> f'_h`, specified per-validator, which transforms the fees received - e.g. a validator could (in the future) opt to donate some fraction to the community pool, or the protocol could opt to pay particular validators more (maybe inversely proportional to stake, in a similar vein to my Tendermint proposer weighting ideas ;) )

Reply:
Great idea! Does the updated description at the end of this subsection sufficiently explain this?